src/main/common/huffman.h

```diff
@@ -24,8 +24,8 @@
 
 #define HUFFMAN_TABLE_SIZE 257 // 256 characters plus EOF
 typedef struct huffmanTable_s {
-    uint8_t     codeLen;
-    uint16_t    code;
+    uint16_t    code:12;
+    uint16_t    codeLen:4;
```

Comment:
Why does it need 16 bits now and `:4` is new to me :heart_eyes:

Reply:
Allocate `uint16_t` , the first `12bits`  will be code  and next `4bits`  will be `codeLen`
And as I have noticed that last 4bits are always zero in `code`  we can reuse them for `codeLen`  as it is max 4bits 
and then we can just shift it `code << 4`